lib/random.gd

```diff
@@ -69,9 +69,86 @@ DeclareCategory( "IsRandomSource", IsComponentObjectRep );
 ##  </ManSection>
 ##  <#/GAPDoc>
 ##
-DeclareOperation( "Random", [IsRandomSource, IsList] );
+DeclareOperation( "Random", [IsRandomSource, IsListOrCollection] );
 DeclareOperation( "Random", [IsRandomSource, IsInt, IsInt] );
 
+##############################################################################
+##
+##  <#GAPDoc Label="InstallMethodWithGlobalRandomGen">
+##  <ManSection>
+##  <Func Name="InstallMethodWithGlobalRandomGen"
+##   Arg="opr,info[,famp],args-filts[,val],method"/>
+##
+##  <Description>
+##  This function is designed to simplify adding new overloads for
+##  <Ref Oper="Random"/> and <Ref Oper="PseudoRandom"/> to GAP which can
+##  be called both with, and without, a random soure.
+##  <P/>
+##  It accepts the same arguments as <Ref Func="InstallMethod"/>, with
+##  the extra requirement that the first member of <A>args-filts</A> must
+##  be IsRandomSource, and the <A>info</A> argument
+##  is compulsory and must begin 'for a random source and'.
+##  <P/>
+##  This function calls <Ref Func="InstallMethod"/> twice. Firstly
+##  with it's arguments unchanged, and secondly with the first argument
+##  removed, instead always passing in the value
+##  <Ref Var="GlobalRandomSource"/>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+BIND_GLOBAL("InstallMethodWithGlobalRandomGen", function(args...)
+    local str, filterpos, filtercopy, argscopy, i, func, info;
+
+    # Check we understand arguments
+    # Second value must be an info string
+    if not IsString(args[2]) then
+        ErrorNoReturn("Second argument must be an info string");
+    fi;
+
+    # Info strings always tend to begin 'for ', and here we want
+    # to be able to edit it, so we check.
+    if args[2]{[1..23]} <> "for a random source and" then
+        ErrorNoReturn("Info string must begin 'for a random source and'");
+    fi;
+
+    # Filters must start with 'IsRandomSource'
+    for i in [1..Length(args)] do
+        if IsList(args[i]) and args[i][1] = IsRandomSource then
+            filterpos := i;
+        fi;
+    od;
+
+    if not IsBound(filterpos) then
+        ErrorNoReturn("Must use a list of filters beginning 'IsRandomSource'");
+    fi;
+
+    # Last argument must be the actual method
+    if not IsFunction(args[Length(args)]) then
+        ErrorNoReturn("Argument list must end with the method");
+    fi;
+
+    # Install 
+    CallFuncList(InstallMethod, args);
+
+    # Install random, wrapping random source argument
+    argscopy := List(args);
+
+    # Remove 'IsRandomSource' from the filter list
+    argscopy[filterpos] := argscopy[filterpos]{[2..Length(argscopy[filterpos])]};
+
+    # Correct info string by removing 'a random source and'
+    info := "for";
+    APPEND_LIST(info, argscopy[2]{[24..Length(argscopy[2])]});
+    argscopy[2] := info;
+
+    func := argscopy[Length(argscopy)];
+    argscopy[Length(argscopy)] := x -> func(GlobalMersenneTwister,x);
```

Comment:
Ahh, so this actually only works if the mtehod takes exactly one argument in addition to the random source. But there are some `Random` methods which take 1+2 = 3 argument. 

If the user tries to use `InstallMethodWithGlobalRandomGen` to set one of those up, the mistake will not be immediately visible -- only if you try to use `Random` without a source. So I would suggest we check the number of arguments. In addition, we could also verify that `NumberArgumentsFunction(func)` and `Length(args[filterpos])` coincide.

Even nicer would of course be if we could support arbitrary numbers of arguments :-). But I am not going to hold this PR hostage over this :-).